qiskit/algorithms/optimizers/qnspsa.py

```diff
@@ -95,6 +92,7 @@ def __init__(
         lse_solver: Optional[Callable[[np.ndarray, np.ndarray], np.ndarray]] = None,
         initial_hessian: Optional[np.ndarray] = None,
         callback: Optional[CALLBACK] = None,
+        termination_checker: Optional[TERMINATIONCHECKERCALLBACK] = None,
```

Comment:
`TERMINATIONCHECKERCALLBACK` is a bit difficult too read being all uppercase and very long. Can we change it for something more legible, maybe `TERMINATION_CHECK` or `TERMINATION_CRITERION`?

Reply:
@Cryoris  I was not happy with TERMINATIONCHECKERCALLBACK  either. I had TERMINATIONCHECKER_CALLBACK (and a variation) , but it was not accepted by the linter 
```
 "TERMINATIONCHECKER_CALLBACK" doesn't conform to '[A-Z_][a-zA-Z0-9]+$' pattern (invalid-name)
```
And `TERMINATION_CHECK` is not accepted either: https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=31165&view=logs&j=f0ab2c38-58fa-55ad-8e8b-d3984171f737&t=980625fc-4a64-50e4-7ca0-679c2a4bbda8
```
qiskit/algorithms/optimizers/spsa.py:33:0: C0103: Class name "TERMINATION_CHECKER" doesn't conform to '[A-Z_][a-zA-Z0-9]+$' pattern (invalid-name)
```
I don't understand the reason though, it seems to match the regular expression.

Reply:
Settled on `TERMINATIONCHECKER`. Underscores in the middle of a name are not allowed by the linter